### DISCUSSION TITLE
Form Context - Field option for Debounce

### DISCUSSION THREAD

**Discussion by mnts-i** [mantinedev/mantine#2575] [Feature requests]
I really like the built-in Form Context and you can create forms really fast and with few lines of code. Would it be possible to implement in the future an option to toggle value debouncing on change for specific fields. Something like this:

```typescript
<TextInput
    label="Last Name"
    variant="filled"
    // Debounce delay. If it's left undefined update the value immediately (current behavior)
    {...form.getInputProps('lastName', { debounce: 200 )}
/>
```

**Comment by rtivital** [comment]
This will lead to unresponsive UI, TextInput value will be updated 200ms after user entered a character.

**Reply by mnts-i** [reply]
I suggested this because everytime I change something on one field, every other field has to rerender (the Context changes and thus it propagates the render to its children) and as a result there is a small (but noticable) lag on each keystroke (form with ~10 fields). A debounce would prevent that. 

After some code digging I found another possible solution. The reason the other fields are getting rerendered is because their onChange and onFocus callbacks change all the time. I assume the ``{...form.getInputProps(...)}`` part always results into a new payload object.  I think (maybe I am wrong) if you wrap the ``getInputProps: GetInputProps<Values>`` and its functions dependencies (such as ``clearFieldError`` function)  inside a useCallback (``mantine-form/src/use-form.tsx`` source file) it will fix this behavior.

**Reply by rtivital** [reply]
Nope, it will not fix the issue, all fields rerender because form state changes. We already went the path with callbacks memoization, it did not work.

**Reply by bankangle** [reply]
Is there/what is the correct way to debounce form values change in uncontrolled mode? I want to make some kind of draft/autosave on backend after some changes are made, but form is not submitted yet
